cmd/kubectl_token_test.go

```diff
@@ -0,0 +1,114 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	apiv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_getAuthProviders(t *testing.T) {
+
+	setupServer := func(response string) *httptest.Server {
+		return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprint(w, response)
+		}))
+	}
+
+	tt := []struct {
+		name              string
+		server            *httptest.Server
+		expectedProviders []TypedProvider
+		expectedErr       string
+	}{
+		{
+			name:   "response ok",
+			server: setupServer(responseOK),
+			expectedProviders: []TypedProvider{
+				&apiv3.AzureADProvider{
+					AuthProvider: apiv3.AuthProvider{
+						Type: "azureADProvider",
+					},
+					RedirectURL: "https://login.microsoftonline.com/258928db-3ed6-49fb-9a7e-52e492ffb066/oauth2/v2.0/authorize?client_id=56168f69-a732-48e2-aa21-8aa0909d0976&redirect_uri=https://rancher-local.enrichman.it/verify-auth-azure&response_type=code&scope=openid",
```

Comment:
Will the tests fail if the `...enrichman.it` domain does not exist ?
If yes, how can we make this test independent of your machines ?
If not, could a placeholder be used which does not refer to your machines ?

Reply:
Hm, no, it will not fail because it's just a static check, but still it's better to avoid references to local machine. It's better to have something more generic. :+1:

Reply:
Done with f50f954